addon/instance-initializers/setup-session-restoration.js

```diff
@@ -1,13 +1,11 @@
-import Ember from 'ember';
-
 export default function setupSessionRestoration(instance) {
   const { container } = instance;
   const applicationRoute = container.lookup('route:application');
   const session = container.lookup('simple-auth-session:main');
   applicationRoute.reopen({
-    beforeModel(transition) {
-      const superCall = Ember.run.bind(this, this._super, transition);
-      return session.restore().then(superCall, superCall);
+    beforeModel() {
+      const superResult = this._super(...arguments);
+      return session.restore().then(() => superResult, () => superResult);
```

Comment:
If `finally` is used, and the session restore promise rejects, the transition is canceled because the rejected promise is returned by `beforeModel`. Handling the error state with `then` works.